src/app/derive/derive.component.ts

```diff
@@ -1,93 +1,110 @@
-      }
-      if (params.expirationDays) {
-        const numDays = parseInt(params.expirationDays, 10);
-        if (numDays > 0) {
-          this.expirationDays = numDays;
+    // first grab the query params
+    this.activatedRoute.queryParams
+      .pipe(
+        switchMap((params) => {
+          // verify they sent the correct parameter permutation
+          this.validationErrors = this.getParameterValidationErrors(params);
+          if (this.validationErrors) {
+            return throwError('invalid query parameter permutation');
+          }
+          if (params.publicKey) {
+            this.publicKeyBase58Check = params.publicKey;
+            this.isSingleAccount = true;
+          }
+          if (params.derivedPublicKey) {
+            this.derivedPublicKeyBase58Check = params.derivedPublicKey;
+          }
+          // We can only revoke if we have both the public key and derived public key from query params.
+          if (
+            params.deleteKey &&
+            this.publicKeyBase58Check &&
+            this.derivedPublicKeyBase58Check
+          ) {
+            this.deleteKey = params.deleteKey === 'true';
+            // We don't want or need to parse transaction spending limit when revoking derived key,
+            // so we initialize a spending limit object with no permissions.
+            this.transactionSpendingLimitResponse = { GlobalDESOLimit: 0 };
+            // Setting expiration days to 0 forces us to have a minimum transaction size that is still valid.
+            this.expirationDays = 0;
+            return of(null);
+          }
+          if (params.transactionSpendingLimitResponse) {
+            this.transactionSpendingLimitResponse = JSON.parse(
+              decodeURIComponent(params.transactionSpendingLimitResponse)
+            );
+          }
+          if (params.expirationDays) {
+            const numDays = parseInt(params.expirationDays, 10);
+            if (numDays > 0) {
+              this.expirationDays = numDays;
+            }
+          }
+          return this.backendApi.GetUserProfiles(publicKeys);
+        })
+      )
+      .subscribe((profiles) => {
+        if (profiles) {
+          // if there's an account in the query params display only that
+          if (this.publicKeyBase58Check) {
+            this.allUsers = {
+              [this.publicKeyBase58Check]: profiles[this.publicKeyBase58Check],
+            };
+          } else {
+            // otherwise display all accounts
+            this.allUsers = profiles;
+          }
         }
-      }
-    });
+      });
     // Set derive to true
     this.globalVars.derive = true;
   }
 
-  redirectLoadSeed(): void {
-    this.router.navigate(['/', RouteNames.LOAD_SEED], { queryParamsHandling: 'merge' });
-  }
-
-  redirectSignUp(): void {
-    this.router.navigate(['/', RouteNames.SIGN_UP], { queryParamsHandling: 'merge' });
-  }
-
-  launchGoogle(): void {
-    this.googleDrive.launchGoogle();
-  }
-
```

Comment:
So there's no more "signup/login with ..." buttons in the derive screen?

Reply:
Yeah we were talking either in slack or on a call where we thought it would be better to remove them. Maybe I misunderstood though